plugin/src/test/java/dev/regadas/trino/pubsub/listener/PubSubEventListenerTest.java

```diff
@@ -0,0 +1,259 @@
+        var eventListener = new PubSubEventListener(config, publisher);
+
+        eventListener.queryCreated(TestData.FULL_QUERY_CREATED_EVENT);
+
+        assertThat(publisher.lastPublishedMessage, instanceOf(Schema.QueryCreatedEvent.class));
+    }
+
+    @Test
+    void testQueryCompletedPublishCorrespondingMessage() {
+        var publisher = new SuccessPublisher();
+        var config = new PubSubEventListenerConfig(true, true, true, "", "", null, Encoding.JSON);
+        var eventListener = new PubSubEventListener(config, publisher);
+
+        eventListener.queryCompleted(TestData.FULL_QUERY_COMPLETED_EVENT);
+
+        assertThat(publisher.lastPublishedMessage, instanceOf(Schema.QueryCompletedEvent.class));
+    }
+
+    @Test
+    void testSplitCompletedPublishCorrespondingMessage() {
+        var publisher = new SuccessPublisher();
+        var config = new PubSubEventListenerConfig(true, true, true, "", "", null, Encoding.JSON);
+        var eventListener = new PubSubEventListener(config, publisher);
+
+        eventListener.splitCompleted(TestData.FULL_SPLIT_COMPLETED_EVENT);
+
+        assertThat(publisher.lastPublishedMessage, instanceOf(Schema.SplitCompletedEvent.class));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"success", "throwOnClose"})
+    void testClosedNormally(String pubType) {
+        var publisher = TestPublisher.from(pubType);
+        var config = new PubSubEventListenerConfig(true, true, true, "", "", null, Encoding.JSON);
+        var eventListener = new PubSubEventListener(config, publisher);
+
+        assertDoesNotThrow(() -> eventListener.close());
+        assertThat(publisher.closeCalled, is(true));
+    }
+
+    public static <T> void assertThatEventually(Supplier<T> actual, Matcher<? super T> matcher)
+            throws InterruptedException {
+        int tries = RETRY_COUNT;
+        while (--tries > 0 && !matcher.matches(actual.get())) {
+            Thread.sleep(RETRY_DELAY);
+        }
+        assertThat(actual.get(), matcher);
+    }
+
+    abstract static class TestPublisher implements Publisher {
+        Message lastPublishedMessage;
+        boolean closeCalled;
+
+        @Override
+        public CompletableFuture<String> publish(Message message) {
+            lastPublishedMessage = message;
+            return doPublish(message);
+        }
+
+        @Override
+        public void close() throws Exception {
+            closeCalled = true;
+            doClose();
+        }
+
+        static TestPublisher from(String pubType) {
+            return switch (pubType) {
+                case "success" -> new SuccessPublisher();
+                case "returnFailure" -> new ReturnFailurePublisher();
+                case "throwOnPublish" -> new ThrowOnPublishPublisher();
+                case "throwOnClose" -> new ThrowOnClosePublisher();
+                default -> throw new IllegalArgumentException("invalid pudType: " + pubType);
```

Comment:
no need for `default`. That's the default behaviour now with the new `switch`

Reply:
You need the default unless the compiler can infer completeness, If I would have done some enums, then the compiler can know that there is no other possible value